### ISSUE TITLE
Functions

### ISSUE THREAD

**Issue by AlexTriada** [SkaceKamen/vscode-sqflint#56]
Great module, I really like it, man, you can deal with the functions, or suggest how to prescribe them in .hpp so that they do not create notifications of type "Possibly undefined variable some_fnc_name_function" - it's very distracting

**Comment by SkaceKamen** [comment]
Functions need to be defined in description.ext (or the definition must be included there) and the expected format is described here:
https://community.bistudio.com/wiki/Arma_3_Functions_Library#Adding_a_Function

there is also limited support for macros, so there's posibility your functions wont get recognized properly if the definitions rely on macros.

**Comment by AlexTriada** [comment]
Functions define in file functions.hpp, mission https://github.com/AlexTriada/A3-AntiStaki

**Comment by Wyqer** [comment]
**Similar to:** #20 
**Current Version:** 0.10.0

**Description**
We're also getting the "possibly undefined variable" on each of our own created functions, which worked fine (even with header as tooltip for parameters, etc.) some versions ago.

**Way to reproduce:**
* Clone <https://github.com/KillahPotatoes/KP-Liberation>
* Switch to branch *v0.97S13*
* Open cloned repo at root directory (right click KP-Liberation folder, Open with Code)
* Wait until the indexing is finished
* You'll get about 645 warnings about possibly undefined own functions

**Structure of project explained**
* In the repository is the `missionframework` folder, which will be the root folder of a mission.
* So there is the important `description.ext` located, which includes here [Link To File](https://github.com/KillahPotatoes/KP-Liberation/blob/v0.97S13/Missionframework/description.ext#L218-L220) our [overall CfgFunctions](https://github.com/KillahPotatoes/KP-Liberation/blob/v0.97S13/Missionframework/KPLIB_functions.hpp)
* That includes the functions of each module like this [module function declaration](https://github.com/KillahPotatoes/KP-Liberation/blob/v0.97S13/Missionframework/modules/01_common/functions.hpp)
* So a function like [KPLIB_fnc_common_createUnit](https://github.com/KillahPotatoes/KP-Liberation/blob/v0.97S13/Missionframework/modules/01_common/fnc/fn_common_createUnit.sqf) is declared.
* In older version this worked without any issues, even the header was parsed right and gave nice parameter and example tooltips on the functions in the code. We have this issues like maybe 2 or 3 versions, when the #52 came up.

**Screenshot**
![warnings](https://user-images.githubusercontent.com/3811977/59155864-e34ee400-8a91-11e9-94ee-9af9a2d6c6c9.png)

**Comment by SkaceKamen** [comment]
Thanks for detailed info, I'll look into it when I have time

**Comment by SkaceKamen** [comment]
@Wyqer: This issue with your description.ext is usage of __EVAL()
That is really complicated macro in terms of parsing ... I'll try to work around it.

**Comment by Wyqer** [comment]
Ok, good to know the possible cause of the issue.
So when you've removed the __EVAL(), it was fine so far?
We've a similar issue also on the old framework. There it seems that the indexing can't finish fully and therefore the whole linting and the functions/commands tooltips aren't working.
If you've some time to spare, you could have a look at it with the **master branch** of the liberation repository. Maybe you find something useful to improve anything.
But in the end the old framework isn't that important and it could also be just bad code which causes the issue, even if it worked fine earlier also in the old framework.

In any case many thanks for your support and work on sqflint, you're making our work so much easier.

**Comment by SkaceKamen** [comment]
Yes, when you remove the __EVAL, it's fine. I've already wrote a fix and will publish it soon, I just want to test if it doesn't break anything else.

@AlexTriada Your description.ext has multiline macros, which aren't supported yet, let me see if I can do anything about it.

**Comment by SkaceKamen** [comment]
@AlexTriada sorry, but the usage of macros in your description.ext (and correlating hpp files) is way too complicated right now, maybe when I have time to completely rewrite the preprocessor :'(

@Wyqer the issue with __EVAL should be fixed now (0.10.1)

**Comment by Wyqer** [comment]
Works perfect now again. Just some other thing, but I'll create a new issue for that. Thanks for adding the feature to support the __EVAL() in description.ext
![function](https://user-images.githubusercontent.com/3811977/59301566-2f687700-8c92-11e9-8767-d908e2ddde9c.png)

**Comment by Wyqer** [comment]
Edit, the leftover now is just what I reported in #55

**Comment by AlexTriada** [comment]
@SkaceKamen Then tell me, are macros not supported at all?

**Comment by jonathan-richer** [comment]
Are the functions getting parsed in the config.cpp?

![image](https://user-images.githubusercontent.com/36356746/77714551-a5cf2f00-6faf-11ea-9069-93f3bd55efd1.png)

```c++
class CfgFunctions {
	class JRS_VehicleDeploy {
		tag = "JRS";
		class vehicle {
			file = "\jrs_vehicle_deploy\functions\vehicle";
			class deployObjects { };
			class initVehicle { };
		};
		class utils {
			file = "\jrs_vehicle_deploy\functions\utils";
			class createObject { };
		};
	};
};
```
